res/controllers/Hercules-DJControl-Inpulse-500-script.js

```diff
@@ -0,0 +1,1566 @@
+    midi: [0x90 + midiChannel, 0x02],
+    type: components.Button.prototype.types.toggle,
+    shiftOffset: 3,
+    shiftControl: false,
+    shiftChannel: true,
+    sendShifted: true,
+    outKey: 'quantize',
+    unshift: function () {
+      this.inKey = 'quantize';
+    },
+    shift: function () {
+      this.inKey = 'keylock';
+    },
+  });
+
+  // Knobs
+  this.volume = new components.Pot({
+    midi: [0xB0 + midiChannel, 0x00],
+    inKey: 'volume',
+  });
+
+  this.eqKnob = [];
+  for (var k = 1; k <= 3; k++) {
+    this.eqKnob[k] = new components.Pot({
+      midi: [0xB0 + midiChannel, 0x01 + k],
+      group: '[EqualizerRack1_' + this.currentDeck + '_Effect1]',
+      inKey: 'parameter' + k,
+    });
+  }
+
+  this.gainKnob = new components.Pot({
+    midi: [0xB0 + midiChannel, 0x05],
+    key: 'pregain',
+  });
+
+  // Pitch-tempo fader
+  this.pitchFader = new components.Pot({
+    midi: [0xB0 + midiChannel, 0x08],
+    key: 'rate',
+  });
+
+  // Jog Wheel
+  // TODO: Handle with less repeat the shift key for this
+  this.jogWheel = new components.JogWheelBasic({
+    midi: [0xB0 + midiChannel, 0x0A],
+    deck: midiChannel, // whatever deck this jogwheel controls, in this case we ignore it
+    wheelResolution: 720, // how many ticks per revolution the jogwheel has
+    alpha: 5/6,
+    beta: (5/6)/128,
+    rpm: 33 + 1/3,
+    group: "[Channel"+midiChannel+"]",
+    inputWheel: function(_channel, _control, value, _status, group) {
+      var deck = parseInt(deckData.currentDeck.charAt(8));
+      value = this.inValueScale(value);
+      if (engine.isScratching(deck)) {
+        engine.scratchTick(deck, value);
+      } else {
+        engine.setValue(group, 'jog', value);
+      }
+    },
+    inputTouch: function(channel, control, value, status, group) {
+      var deck = parseInt(deckData.currentDeck.charAt(8));
+      if ((value === 0x7F) && deckData.vinylButtonState[deck - 1]) {
+        engine.scratchEnable(deck,
+          this.wheelResolution,
+          this.rpm,
+          this.alpha,
+          this.beta);
+      } else {
+        engine.scratchDisable(deck);
+      }
+    },
+  });
+
+  this.jogWheelShift = new components.JogWheelBasic({
+    midi: [0xB3 + midiChannel, 0x0A],
+    deck: midiChannel, // whatever deck this jogwheel controls, in this case we ignore it
+    wheelResolution: 720, // how many ticks per revolution the jogwheel has
+    alpha: 5/6,
+    beta: (5/6)/128,
+    rpm: 33 + 1/3,
+    group: "[Channel"+midiChannel+"]",
+    inputWheel: function(_channel, _control, value, _status, group) {
+      var deck = parseInt(deckData.currentDeck.charAt(8));
+      value = this.inValueScale(value) * 4;
+      if (engine.isScratching(deck)) {
+        engine.scratchTick(deck, value);
+      } else {
+        engine.setValue(group, 'jog', value);
+      }
+    },
+    inputTouch: function(channel, control, value, status, group) {
+      var deck = parseInt(deckData.currentDeck.charAt(8));

```

Comment:
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
      var deck = script.deckFromGroup(deckData.currentDeck);
```